lib/sensors/MoistureSensor.cpp

```diff
@@ -0,0 +1,93 @@
+#include "MoistureSensor.h"
+
+constexpr int Sensor_I2C_Address = 0x36 << 1;                               //MBED I2C uses 8 bit addressing, so addresses are left shifted by 1 (may need to be shifted by 2) 
+
+constexpr int Sensor_Status_Base = 0x00;                                    //Base address registers for different modules
+constexpr int Sensor_Moisture_Base = 0x0F;
+
+constexpr int Sensor_Status_HW_ID = 0x01;                                   //Function address register for the sensor's HW ID
+constexpr int Sensor_HW_ID_Code = 0x55;                                     //Expected value for sensor HW ID
+
+constexpr int Sensor_Moisture_Function = 0x10;                              //Function address registers for various modules       
+constexpr int Sensor_Temp_Function = 0x04;  
+constexpr int Sensor_Status_Reset = 0x7F;
+
+
+MoistureSensor::MoistureSensor(PinName sda, PinName scl) : i2c_(sda, scl){}
+
+bool MoistureSensor::Is_Initialized(){
+    return (this->Read_HW_ID() == Sensor_HW_ID_Code);                       //compare received HW ID Code to correct one
+}
+
+void MoistureSensor::Reset_Sensor(){
+    char cmd[3];
+    cmd[0] = Sensor_Status_Base;                                            //initialize registers for clearing sensor memory
+    cmd[1] = Sensor_Status_Reset;
+    cmd[2] = 0xFF;
+
+    i2c_.write(Sensor_I2C_Address, cmd, 3);                                 //set all registers on sensor to default values
+}
+
+uint8_t MoistureSensor::Read_HW_ID(){
+    char cmd[2];
+    cmd[0] = Sensor_Status_Base;
+    cmd[1] = Sensor_Status_HW_ID;
+
+    char check[1];
+
+    i2c_.write(Sensor_I2C_Address, cmd, 2);                                 //initialize registers for checking device ID
+    ThisThread::sleep_for(125);
+    i2c_.read(Sensor_I2C_Address, check, 1);                                //read device ID
+
+    return check[0];
+}
+
+uint16_t MoistureSensor::Read_Moisture(){
+    if(!(this->Is_Initialized())){                                          //checks if device is initialized, returns 65534 if there is an issue
```

Comment:
If this is being read infrequently this is alright, but if it is being read frequently I would recommend calling the initialization function once in the science main.cpp before looping and setting a private boolen in this class to be read by the Read and similar functions instead of reading the register repeatedly.

Reply:
Please summarize the issues that you ran into while testing the sensor, I remember you mentioning weird things going on with the addresses and Hw ID but I don't remember if they were fixed... You can merge this PR as-is, but please document those quirks as new GitHub issues.

Reply:
The frequency of the reading will depend on how science team chooses to implement it; if it ends up being a problem going forward, I'll restructure the code to avoid any issues.

As for the latter comment, those issues were indeed fixed, and I was able to get all required functionality working :D